Group Chats Docs (#650)
xmtp/xmtp-dot-org: docs/build/groups.md

</TabItem>
</Tabs>

## Send a message in a group chat

Send a message to an existing group chat. Refer to the [Messages](/docs/build/messages) section for more details.

<Tabs groupId="groupchats">
<TabItem value="kotlin" label="Kotlin" attributes={{className: "kotlin_tab"}}>

```kotlin
val group = client.conversations.newGroup(listOf(walletAddress1,walletAddress2))
//Send a message
group.send("Hello, group!")
```

</TabItem>
<TabItem value="swift" label="Swift"  attributes={{className: "swift_tab"}}>

```swift
let group = try await client.conversations.newGroup(with: [walletAddress1, walletAddress2])
//Send a message
try await group.send(content: "Hello, group!")
```

</TabItem>
<TabItem value="rn" label="React Native" attributes={{className: "rn_tab "}}>

Code sample coming soon

</TabItem>
<TabItem value="dart" label="Dart"  attributes={{className: "dart_tab"}}>

Code sample coming soon

</TabItem>
<TabItem value="js" label="JavaScript"  attributes={{className: "js_tab "}}>

Code sample coming soon

</TabItem>
<TabItem value="react" label="React"  attributes={{className: "react_tab "}}>

Code sample coming soon

</TabItem>
</Tabs>

## Manage group chat members

> Here are the ways that you manage group chat members with XMTP.

Reviewer: Is this TBD?

### List group members

Retrieve a list of wallet addresses for all members in the group chat

<Tabs groupId="groupchats">
<TabItem value="kotlin" label="Kotlin" attributes={{className: "kotlin_tab"}}>

```kotlin
group.sync()
val members = group.memberAddresses()
```

</TabItem>
<TabItem value="swift" label="Swift"  attributes={{className: "swift_tab"}}>

```swift
try await group.sync()
let members = group.members()
```

</TabItem>
<TabItem value="rn" label="React Native" attributes={{className: "rn_tab "}}>

Code sample coming soon

</TabItem>
<TabItem value="dart" label="Dart"  attributes={{className: "dart_tab"}}>

Code sample coming soon

</TabItem>
<TabItem value="js" label="JavaScript"  attributes={{className: "js_tab "}}>

Code sample coming soon

</TabItem>
<TabItem value="react" label="React"  attributes={{className: "react_tab "}}>

Code sample coming soon

</TabItem>
</Tabs>

### Add group members

> Add new members to an existing group chat using its wallet address.

Reviewer: ```suggestion
Add new members to an existing group chat using their wallet addresses.
```

<Tabs groupId="groupchats">
<TabItem value="kotlin" label="Kotlin" attributes={{className: "kotlin_tab"}}>

```kotlin
group.addMembers(listOf(walletAddress))
```

</TabItem>
<TabItem value="swift" label="Swift"  attributes={{className: "swift_tab"}}>

```swift
try await group.addMembers(addresses: [walletAddress])
```

</TabItem>
<TabItem value="rn" label="React Native" attributes={{className: "rn_tab "}}>

Code sample coming soon

</TabItem>
<TabItem value="dart" label="Dart"  attributes={{className: "dart_tab"}}>

Code sample coming soon

</TabItem>
<TabItem value="js" label="JavaScript"  attributes={{className: "js_tab "}}>

Code sample coming soon

</TabItem>
<TabItem value="react" label="React"  attributes={{className: "react_tab "}}>

Code sample coming soon

</TabItem>
</Tabs>

### Remove group members

> Remove a member from an existing group chat using its wallet address

Reviewer: ```suggestion
Remove members from an existing group chat using their wallet addresses
```

This is based on the title of the section being plural and assuming the function takes more than one.

:::caution Group Chats are Per Installation
Group chats in XMTP are specific to each installation. This means that while you will see your group chat conversations across different devices, you will not see the historical messages within those chats automatically. Each group chat's message history is tied to the device where it was started. Consequently, there is no message history synced across devices. When you sign in on a new device, you will be able to see existing group chat conversations but will only receive new messages going forward. This behavior is specific to group conversations.
:::

## Create a group chat

Initiate a new group chat with a list of specified addresses. To create a group, the recipient must have already started their client at least once on the XMTP network.

<Tabs groupId="groupchats">
<TabItem value="rn" label="React Native" attributes={{className: "rn_tab "}}>

```jsx
const group = await client.conversations.newGroup([
  walletAddress1,
  walletAddress2,

> ]);

Reviewer 2: You could call out the new admin permissions we just shipped as well.

You can now pass a permissions field to newGroup that can take a enum of 
everyone_is_admin
creator_is_admin

Which either allows everyone to add and remove members or only the group creator to do that.

```

</TabItem>
<TabItem value="swift" label="Swift"  attributes={{className: "swift_tab"}}>

```swift
let group = try await client.conversations.newGroup(with: [walletAddress1, walletAddress2])
```

</TabItem>
<TabItem value="dart" label="Dart"  attributes={{className: "dart_tab"}}>

Code sample expected for Q2 2024

</TabItem>
<TabItem value="js" label="JavaScript"  attributes={{className: "js_tab "}}>

Code sample expected for Q2 2024

</TabItem>
<TabItem value="react" label="React"  attributes={{className: "react_tab "}}>

Code sample expected for Q2 2024

</TabItem>
</Tabs>

_The maximum amount of addresses allowed is 250._

## List group chat conversations

Retrieve all existing group chat conversations associated with the current XMTP client. Refer to the [Conversations](/docs/build/conversations.md) section for more details.

<Tabs groupId="groupchats">
<TabItem value="rn" label="React Native" attributes={{className: "rn_tab "}}>

```jsx
const groups = await client.conversations.listGroups();
```

</TabItem>
<TabItem value="kotlin" label="Kotlin" attributes={{className: "kotlin_tab"}}>

```kotlin
// List all group chats for the current user

> val groups = client.conversations.listGroups()

Reviewer 2: Might be a good place to call out client.conversations.syncGroups() since the list groups will just pull whatever is in the database. So if you want the latest sync before listing to kick off a network request for the latest groups.

</TabItem>
<TabItem value="react" label="React"  attributes={{className: "react_tab "}}>

Code sample expected for Q2 2024

</TabItem>
</Tabs>

## Manage group chat members

You can list, add and remove members from a group chat. The current limit is 250. Only the creator of the group chat has the permissions to add or remove members. This restriction ensures that only authorized individuals can modify the participant list. Developers should design their application's user interface and functionality with this consideration in mind, ensuring that options to add or remove members are only available to the group's creator.

### List group members

Retrieve a list of wallet addresses for all members in the group chat

<Tabs groupId="groupchats">
<TabItem value="rn" label="React Native" attributes={{className: "rn_tab "}}>

```jsx
await group.sync();
const members = await group.memberAddresses();
```

</TabItem>
<TabItem value="kotlin" label="Kotlin" attributes={{className: "kotlin_tab"}}>

```kotlin
group.sync()
val members = group.memberAddresses()
```

</TabItem>
<TabItem value="swift" label="Swift"  attributes={{className: "swift_tab"}}>

```swift
try await group.sync()

> let members = group.members()

Reviewer 2: This got renamed to `memberAddresses`
```suggestion
let members = group.memberAddresses()
```
